nexus-repository-composer/src/main/java/org/sonatype/nexus/repository/composer/internal/ComposerProxyFacetImpl.java

```diff
@@ -103,18 +103,24 @@ protected Content getCachedContent(final Context context) throws IOException {
   protected Content store(final Context context, final Content content) throws IOException {
     AssetKind assetKind = context.getAttributes().require(AssetKind.class);
     switch (assetKind) {
-      case PACKAGES:
-        return content().put(PACKAGES_JSON, generatePackagesJson(context), assetKind);
-      case LIST:
-        return content().put(LIST_JSON, content, assetKind);
-      case PROVIDER:
-        return content().put(buildProviderPath(context), content, assetKind);
-      case PACKAGE:
-        return content().put(buildPackagePath(context), content, assetKind);
-      case ZIPBALL:
-        return content().put(buildZipballPath(context), content, assetKind);
-      default:
-        throw new IllegalStateException();
+    case PACKAGES:
+    	Content returnedContent;
+    	try {
+    		returnedContent = generatePackagesJson(context);
+    	} catch (NullPointerException e) {
+    		returnedContent=content;
```

Comment:
```suggestion
    		returnedContent = content;
```

Reply:
Service the original package.json as cached artifact might work, if the upstream repository uses the same absolute paths, e.g `/p2/%package%.json`.

Otherwise this pass-through of the original _package.json_ might lead to broken repositories, e.g. proxying packages.drupal.org with a `/files/packages/8/p2/%package%.json` pattern. This path won't resolve on our Nexus instance.

The URLs should be rewritten in any case.